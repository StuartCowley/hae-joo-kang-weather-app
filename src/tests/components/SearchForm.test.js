import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "Test city",
    setSearchText: jest.fn(),
    onSubmit: jest.fn(),
  };

  it("renders correctly", () => {
    const rendered = renderer.create(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(rendered).toMatchSnapshot();
  });

  it("renders correct text", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(screen.getByTestId("search-form__input")).toHaveValue("Test city");
  });

  it("when search button is clicked, onSubmit is called", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    fireEvent.click(screen.getByRole("button"));
    expect(validProps.onSubmit).toHaveBeenCalled();
  });
});
