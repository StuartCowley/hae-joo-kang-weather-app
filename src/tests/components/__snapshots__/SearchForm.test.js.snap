// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<div
  className="search-form"
>
  <input
    data-testid="search-form__input"
    onChange={[Function]}
    placeholder="city name"
    type="text"
    value="Test city"
  />
  <button
    onClick={[MockFunction]}
    type="submit"
  >
    Search
  </button>
</div>
`;
